src/main/java/org/folio/fqm/repository/EntityTypeRepository.java

```diff
@@ -48,30 +53,45 @@ public class EntityTypeRepository {
   private final DSLContext jooqContext;
   private final ObjectMapper objectMapper;
 
+  @Value("${mod-fqm-manager.entity-type-cache-timeout-seconds:3600}")
+  private long cacheDurationSeconds;
+  private final Cache<String, Map<UUID, EntityType>> entityTypeCache = Caffeine.newBuilder().expireAfterWrite(cacheDurationSeconds, TimeUnit.SECONDS).build();
+
   public Optional<EntityType> getEntityTypeDefinition(UUID entityTypeId, String tenantId) {
     return getEntityTypeDefinitions(Collections.singleton(entityTypeId), tenantId).findFirst();
   }
 
   public Stream<EntityType> getEntityTypeDefinitions(Collection<UUID> entityTypeIds, String tenantId) {
-    String tableName = tenantId == null ? TABLE_NAME : tenantId + "_mod_fqm_manager." + TABLE_NAME;
     log.info("Getting definitions name for entity type ID: {}", entityTypeIds);
 
-    Field<String> definitionField = field(DEFINITION_FIELD_NAME, String.class);
-    Condition entityTypeIdCondition = isEmpty(entityTypeIds) ? trueCondition() : field("id").in(entityTypeIds);
-    return readerJooqContext
-      .select(definitionField)
-      .from(table(tableName))
-      .where(entityTypeIdCondition)
-      .fetch(definitionField)
-      .stream()
-      .map(this::unmarshallEntityType)
-      .map(entityType -> {
-        String customFieldsEntityTypeId = entityType.getCustomFieldEntityTypeId();
-        if (customFieldsEntityTypeId != null) {
-          entityType.getColumns().addAll(fetchColumnNamesForCustomFields(UUID.fromString(customFieldsEntityTypeId), entityType));
-        }
-        return entityType;
-      });
+    Map<UUID, EntityType> entityTypes = entityTypeCache.get(tenantId == null ? "" : tenantId, tenantIdKey -> {
+        String tableName = "".equals(tenantIdKey) ? TABLE_NAME : tenantIdKey + "_mod_fqm_manager." + TABLE_NAME;
```

Comment:
This `""` change is kind of hacky, but you can't have a `null` cache key, and turning `null` into a special value is super easy and doesn't feel quite as gross when the special value replacement is done on the line above where it's handled